creator/html_paragraph.go

```diff
@@ -0,0 +1,625 @@
+				}
+			}
+		}
+	}
+}
+
+// Width returns the width of the Drawable.
+func (t *htmlTable) Width() float64 {
+	t.generateContent()
+	return t.table.Width()
+}
+
+// Height returns the height of the Drawable.
+func (t *htmlTable) Height() float64 {
+	t.generateContent()
+	return t.table.Height()
+}
+
+// GeneratePageBlocks generates the page blocks.  Multiple blocks are generated if the contents wrap
+// over multiple pages. Implements the Drawable interface.
+func (t *htmlTable) GeneratePageBlocks(ctx DrawContext) ([]*Block, DrawContext, error) {
+	t.generateContent()
+	return t.table.GeneratePageBlocks(ctx)
+}
+
+type htmlTableStack struct {
+	tableStack []*htmlTable
+}
+
+func (st *htmlTableStack) createAndPushTable() *htmlTable {
+	t := &htmlTable{}
+	st.pushTable(t)
+	return t
+}
+
+func (st *htmlTableStack) currentTable() *htmlTable {
+	if len(st.tableStack) == 0 {
+		return nil
+	}
+	return st.tableStack[len(st.tableStack)-1]
+}
+
+func (st *htmlTableStack) pushTable(table *htmlTable) {
+	st.tableStack = append(st.tableStack, table)
+}
+
+func (st *htmlTableStack) popTable() *htmlTable {
+	t := st.currentTable()
+	if len(st.tableStack) < 1 {
+		return t
+	}
+	st.tableStack = st.tableStack[:len(st.tableStack)-1]
+	return t
+}
+
+type htmlBlock struct {
+	owner      *HTMLParagraph
+	parent     *htmlBlock
+	tableStack *htmlTableStack
+	styleStack *htmlStyleStack
+
+	elements         []VectorDrawable
+	currentParagraph *StyledParagraph
+	style            htmlBlockStyle
+}
+
+func newHTMLBlock(parent *htmlBlock, style htmlBlockStyle) *htmlBlock {
+	b := htmlBlock{
+		owner:            parent.owner,
+		parent:           parent,
+		tableStack:       parent.tableStack,
+		styleStack:       parent.styleStack,
+		elements:         nil,
+		currentParagraph: nil,
+		style:            style,
+	}
+	return &b
+}
+
+var ignoreReplacer = strings.NewReplacer("\r", "", "\n", "", "\t", " ")
+
+func (b *htmlBlock) parseNodeStyle(node *html.Node) htmlBlockStyle {
+	style := b.styleStack.currentStyle()
+	for _, attr := range node.Attr {
+		switch attr.Key {
+		case "style":
+			style.addEmbeddedCSS(node.Data, attr.Val)
+		case "align":
+			switch attr.Val {
+			case "center":
+				style.TextAlignment = TextAlignmentCenter
+			case "left":
+				style.TextAlignment = TextAlignmentLeft
+			case "right":
+				style.TextAlignment = TextAlignmentRight
+			case "justify":
+				style.TextAlignment = TextAlignmentJustify
+			}
+		}
+	}
+	return style
+}
+
+func (b *htmlBlock) processNode(node *html.Node) error {
+	newB := b
+
+	switch node.Type {
+	case html.TextNode:
+		text := ignoreReplacer.Replace(node.Data)
+		if text == "" {
+			return nil
+		}
+		p, created := b.getCurrentOrCreateParagraph()
+		if created {
+			b.currentParagraph.alignment = b.styleStack.currentStyle().TextAlignment
+		}
+		p.Append(text).Style = b.styleStack.currentStyle().TextStyle
+		return nil
+	case html.ElementNode:
+
+		switch node.Data {
+		case "style":
+			log.Println(node)
```

Comment:
Use `common.Log.Debug` instead of `log.Println` for printing debugging messages.